libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxUpcomingManeuverAdapter.kt

```diff
@@ -77,8 +79,10 @@ class MapboxUpcomingManeuverAdapter(
      * @param maneuverToRemove Maneuver
      */
     fun removeManeuver(maneuverToRemove: Maneuver) {
-        if (this.upcomingManeuverList.contains(maneuverToRemove)) {
-            val index = this.upcomingManeuverList.indexOf(maneuverToRemove)
+        val index = this.upcomingManeuverList.indexOfFirst {
+            it.primary.text == maneuverToRemove.primary.text
+        }
+        if (index != -1) {
```

Comment:
Would (index >= 0) be a little more straight forward?

Reply:
Is there a unit test for this condition?

Reply:
> Would (index >= 0) be a little more straight forward?

I like to use -1 as it's more consistent with what the API has to return in case the item is not in the list.

Reply:
> Is there a unit test for this condition?

Done.

Reply:
> > Would (index >= 0) be a little more straight forward?
> 
> I like to use -1 as it's more consistent with what the API has to return in case the item is not in the list.

Ok, but your statement is saying every value except -1 is valid so -2 would be a valid value. I realize that the statement that producing the 'index' value wouldn't return a value of -2 but the statement doesn't reflect the accurate domain of valid values.  Just a NIT.